server/etcdserver/api/etcdhttp/peer.go

```diff
@@ -63,7 +63,6 @@ func newPeerHandler(
 	mux.Handle(peerMembersPath, peerMembersHandler)
 	mux.Handle(peerMemberPromotePrefix, peerMemberPromoteHandler)
 	if leaseHandler != nil {
-		mux.Handle(leasehttp.LeasePrefix, leaseHandler)
```

Comment:
There isn't any change on the client facing API. This is just for peer communication. But there may be some errors during upgrade when only the leader has already been upgraded.  The renew request to the leader may get a 404 response. 

Once all of them reach to the same version, everything will be working well.

Reply:
Yes, you are right. It's even complicated than what we have thought about. I will provide a more detailed plan on how to smoothly support this, which should be backward & forward compatible. 

Actually the PR [9924](https://github.com/etcd-io/etcd/pull/9924) is a bad example, because it breaks the compatibility when part of the node in 3.5 during upgrading etcd from 3.4 to 3.5. But since the the [checkpoint](https://github.com/etcd-io/etcd/blob/5a7df035929a52edabb8130c63926e6531a371c3/server/etcdmain/config.go#L261) is an experimental feature, and it's disabled by default, so nobody runs into the upgrade compatibility issue so far.

Reply:
https://github.com/etcd-io/etcd/pull/9924 is an interesting case as it's etcd command line configuration that affects etcd state. As you cannot imminently set this flags on all cluster members, it means that different members will have different state. For leases this is not a big deal as this is transient, however we should be careful about such changes. 

Depending on number of such situations we might want to consider having such configuration negotiated via raft. I wanted to propose it some time ago, however this is currently not high on list of my priories, as we have a lot of other problems.

Reply:
> Depending on number of such situations we might want to consider having such configuration negotiated via raft. 

It looks like a good & generic point. We can discuss it separately if needed.

Reply:
I added a new flag `--experimental-enable-lease-v2-renew`, and the default value is `true` in 3.6. It means that etcd will continue to use the legacy renew implementation by default. We can consider to deprecate the legacy renew implementation in 3.7 and set the flag as `false` by default. Eventually we can remove the legacy renew implementation in 3.8.